subsys/suit/ipuc/src/suit_ipuc_sdfw.c

```diff
@@ -0,0 +1,695 @@
+			declared_count++;
+		}
+	}
+	*count = declared_count;
+	k_mutex_unlock(&ipuc_mutex);
+	return SUIT_PLAT_SUCCESS;
+}
+
+suit_plat_err_t suit_ipuc_sdfw_get_info(size_t idx, struct zcbor_string *component_id,
+					suit_manifest_role_t *role)
+{
+	if (component_id == NULL || role == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	size_t cur_idx = 0;
+
+	k_mutex_lock(&ipuc_mutex, K_FOREVER);
+	for (size_t i = 0; i < CONFIG_SUIT_IPUC_SIZE; i++) {
+		suit_ipuc_entry_t *ipuc_entry = &components[i];
+
+		if (ipuc_entry->component_id.value != NULL && ipuc_entry->component_id.len != 0) {
+			if (cur_idx == idx) {
+				component_id->value = ipuc_entry->component_id.value;
+				component_id->len = ipuc_entry->component_id.len;
+				*role = ipuc_entry->role;
+				k_mutex_unlock(&ipuc_mutex);
+				return SUIT_PLAT_SUCCESS;
+			}
+			cur_idx++;
+		}
+	}
+
+	k_mutex_unlock(&ipuc_mutex);
+	return SUIT_PLAT_ERR_NOT_FOUND;
+}
+
+suit_plat_err_t suit_ipuc_sdfw_write_setup(int ipc_client_id, struct zcbor_string *component_id,
+					   struct zcbor_string *encryption_info,
+					   struct zcbor_string *compression_info)
+{
+	intptr_t slot_address = 0;
+	size_t slot_size = 0;
+
+	if (component_id == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	if (ipuc_decode_mem_component_id(component_id, &slot_address, &slot_size) !=
+	    SUIT_PLAT_SUCCESS) {
+		/* Only MRAM-based MEM components are supported
+		 */
+		return SUIT_PLAT_ERR_UNSUPPORTED;
+	}
+
+	if (encryption_info != NULL && encryption_info->value != NULL &&
+	    encryption_info->len != 0) {
+		/* Encryption not supported yet
+		 */
+		return SUIT_PLAT_ERR_UNSUPPORTED;
+	}
+
+	if (compression_info != NULL && compression_info->value != NULL &&
+	    compression_info->len != 0) {
+		/* Compression not supported yet
+		 */
+		return SUIT_PLAT_ERR_UNSUPPORTED;
+	}
+
+	k_mutex_lock(&ipuc_mutex, K_FOREVER);
+	suit_ipuc_entry_t *ipuc_entry = ipuc_entry_from_component_id(component_id);
+
+	if (ipuc_entry == NULL) {
+		k_mutex_unlock(&ipuc_mutex);
+		return SUIT_PLAT_ERR_NOT_FOUND;
+	}
+
+	if (!(ipuc_entry->usage == IPUC_UNUSED || (ipuc_entry->usage == IPUC_IPC_IN_PLACE_UPDATE &&
+						   ipuc_entry->ipc_client_id == ipc_client_id))) {
+		LOG_ERR("suit_ipuc_sdfw_write_setup, IPUC address: %p, busy", (void *)slot_address);
+		k_mutex_unlock(&ipuc_mutex);
+		return SUIT_PLAT_ERR_INCORRECT_STATE;
+	}
+
+	/* Erasing if necessary
+	 */
+	uint8_t read_buffer[READ_CHUNK_MAX_SIZE];
+	struct stream_sink ram_sink;
+
+	if (suit_ram_sink_get(&ram_sink, read_buffer, READ_CHUNK_MAX_SIZE) != SUIT_PLAT_SUCCESS) {
+		LOG_ERR("Could not acquire RAM sink");
+		k_mutex_unlock(&ipuc_mutex);
+		return SUIT_PLAT_ERR_IO;
+	}
+
+	bool clear = true;
+
+	for (size_t off = 0; off < slot_size && clear == true; off += READ_CHUNK_MAX_SIZE) {
+		size_t cur_chunk_size = slot_size - off;
+
+		if (cur_chunk_size > READ_CHUNK_MAX_SIZE) {
+			cur_chunk_size = READ_CHUNK_MAX_SIZE;
+		}
+
+		suit_address_streamer stream_fn =
```

Comment:
Some lines before it is stated that "Only MRAM-based MEM components are supported" - if we manage to enforce that (which I think we currently don't), then isn't using ram_sink and `suit_address_streamer_select_by_address` a bit of a overkill here? We should be simply able to directly verify that all of the memory is 0xFF.